testcases/kernel/syscalls/write/write04.c

```diff
@@ -51,152 +51,126 @@
 #include <setjmp.h>
 #include <errno.h>
 #include <string.h>
-#include "test.h"
+#include <stdio.h>
+#include "tst_test.h"
 
 #define PIPE_SIZE_TEST getpagesize()
 
 void alarm_handler();
-void setup();
-void cleanup();
-
-char *TCID = "write04";
-int TST_TOTAL = 1;
 
 char fifo[100] = "fifo";
 static sigjmp_buf jmp;
 int rfd, wfd;
 
-int main(int argc, char **argv)
+static void verify_write(void)
 {
-	int lc;
-
 	struct stat buf;
 	int fail;
 	int cnt;
 	char wbuf[17 * PIPE_SIZE_TEST];
 	struct sigaction sigptr;	/* set up signal handler */
 
-	tst_parse_opts(argc, argv, NULL, NULL);
+	if (mknod(fifo, S_IFIFO | 0777, 0) < 0)
+		tst_res(TBROK | TERRNO, "mknod() failed, errno: %d", errno);
 
-	/* global setup */
-	setup();
+	if (stat(fifo, &buf) != 0)
+		tst_res(TBROK | TERRNO, "stat() failed, errno: %d", errno);
 
```

Comment:
We do have the SAFE_FOO() macros for most of the syscalls like the stat(), mknod(), etc. can you please make use of them: https://github.com/linux-test-project/ltp/wiki/Test-Writing-Guidelines#224-safe-macros ?

Reply:
yes; I intentionally kept the old code to make sure I didn't introduce accidentally a bug by misusing them, or change the logic (indeed, I also kept code that is buggy, so the commit will be also bug compatible).

do you want me to add commits to this branch that do all those additional changes?, close this and create a new one? or just push --force on top of this with the next RC?

Reply:
For rewriting old buggy testcases I'm fine a single patch per test rewrite. The incremental approach does not make IMHO much sense there, so I don't tend to overcomplicate the git log, but I'm fine with incremental approach as well. So choose what suits you best.